### ISSUE TITLE
TP-LINK Archer T4U on Ubuntu 14.04. ping time spikes exactly every 120 seconds

### ISSUE THREAD

**Issue by yuqian90** [abperiasamy/rtl8812AU_8821AU_linux#24]
I'm using TP-LINK Archer T4U wireless adapter on ubuntu 64 bit with rtl8812AU_8821AU_linux. I cloned the latest source from the repository as of yesterday (6 Sep). I notice that every 120 seconds exactly, I'll experience a spike in the ping time, as high as 4000 ms (normal ping time is about 8ms). I don't know what's going on here. The same machine running Windows 7 with the same wireless adapter using the stock driver for Windows does not have this problem.

``` code
$ lsb_release -a
Distributor ID: Ubuntu
Description:    Ubuntu 14.04.1 LTS
Release:    14.04
Codename:   trusty

$ uname -a
Linux ubuntu 3.13.0-32-generic #57-Ubuntu SMP Tue Jul 15 03:51:08 UTC 2014 x86_64 x86_64 x86_64 GNU/Linux


Log from /var/log/syslog
...
Sep  7 16:00:05 ubuntu kernel: [15084.605532]    ==> rssi_level:2, rate_bitmap:0x0f8ff000
Sep  7 16:01:01 ubuntu kernel: [15140.766059] UpdateHalRAMask8812A => mac_id:0, networkType:0x0b, mask:0x0fffffff
Sep  7 16:01:01 ubuntu kernel: [15140.766059]    ==> rssi_level:1, rate_bitmap:0x0f8f0000
Sep  7 16:03:59 ubuntu kernel: [15319.276129] UpdateHalRAMask8812A => mac_id:0, networkType:0x0b, mask:0x0fffffff
Sep  7 16:03:59 ubuntu kernel: [15319.276129]    ==> rssi_level:2, rate_bitmap:0x0f8ff000
Sep  7 16:04:07 ubuntu kernel: [15327.298639] UpdateHalRAMask8812A => mac_id:0, networkType:0x0b, mask:0x0fffffff
Sep  7 16:04:07 ubuntu kernel: [15327.298639]    ==> rssi_level:1, rate_bitmap:0x0f8f0000
...
```

![plotresults](https://cloud.githubusercontent.com/assets/6637585/4177881/177bcd7c-3666-11e4-8491-72d1dce6f02f.png)

**Comment by akanouras** [comment]
Have you seen this? https://bugzilla.gnome.org/show_bug.cgi?id=513820

TL;DR: It's usually caused by Network Manager's background network scans. Try setting your AP's BSSID in NM to disable roaming between different BSSIDs of your ESSID - should eliminate most of them.

**Comment by ddrozdov** [comment]
@akanouras, I should've seen your comment months ago... Just found the same solution [here](https://bugzilla.redhat.com/show_bug.cgi?id=490493#c16) today and it works fine, no lags anymore!

**Comment by choikwa** [comment]
Hmmm... How can I tell if the problem is due to NM? I'm wondering if the same latency I see is due to this.

**Comment by choikwa** [comment]
I can anecdotally confirm that lag spike issue is gone after setting BSSID from Network Connections.

**Comment by choikwa** [comment]
Sigh, I wish this was backported to 14.04... is there a process to do this? anyone know who to contact?

**Comment by wmbr** [comment]
Just wanted to add that I had the same issue with [rtl8192du](https://github.com/lwfinger/rtl8192du) driver on a 4.2 kernel with NM 1.0.4 and setting the BSSID also seems to help.

@choikwa Set NM loglevel to debug and look if you encounter a log entry `add_pending_action:
scan` exactly when the network latency spikes.
